### PR TITLE
chore: Automate endpoints generation

### PR DESCRIPTION
**Pull request by erezrokah**
Fixes https://github.com/cloudquery/cq-provider-aws/issues/652

### CODE REVIEW

**Comment by erezrokah** [comment]
Example in https://github.com/cloudquery/cq-provider-aws/pull/740 (I fixed the issue that caused commit linting to fail)

File: workflows/repo-specific/cq-provider-aws/endpoints.yml [hunk 1]
```diff
@@ -14,12 +14,21 @@ jobs:
         uses: actions/setup-go@v2
         with:
           go-version: ^1.17
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
```
[the remark this answers is not in the document]
Comment (erezrokah) [reply workflows/repo-specific/cq-provider-aws/endpoints.yml:17]: Good point! How about I do it in a separate PR so I can verify it?
I used the caching action before and sometimes it actually slows things down due to the time it takes to download and extract the cache

File: workflows/repo-specific/cq-provider-aws/endpoints.yml [hunk 2]
```diff
@@ -14,12 +14,21 @@ jobs:
...
-          test "$(git status -s ./client/data/partition_service_region.json | wc -l)" -eq 0
+
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v4
+        with:
+          # required so the PR triggers workflow runs
+          token: ${{ secrets.GH_CQ_BOT }}
+          branch: fix/update_endpoints
+          base: main
+          title: "fix: Update endpoints"
```
[the remark this answers is not in the document]
Comment (erezrokah) [reply workflows/repo-specific/cq-provider-aws/endpoints.yml]: Done in https://github.com/cloudquery/.github/pull/33/commits/66a3d35230a5cdecfa38855f5db48a9cd6c46fa0

File: workflows/repo-specific/cq-provider-aws/endpoints.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (erezrokah) [reply workflows/repo-specific/cq-provider-aws/endpoints.yml:17]: Going to merge and do the caching separately

[the remark this answers is not in the document]
Comment (bbernays) [reply workflows/repo-specific/cq-provider-aws/endpoints.yml:17]: Great point!

**Pull request merged**